### PR TITLE
Add optional TensorBoard output

### LINKED ISSUE #14: idea: Optional logging to `tensorboard`
I've been poking around at the hyperparams a bit, suspecting that some tweaks to the tau warmup/decay could yield some small efficiency improvements. [`tensorboard`](https://www.tensorflow.org/tensorboard) seems super useful for comparing batches of experiments. (spoiler: the values you chose seem really good, lol) 

Implementation shown below is very rough, but could be cleaned up.

https://github.com/user-attachments/assets/8c483764-5846-4212-b1a6-7ec18bb85130

### PR DESCRIPTION
**Pull request by rparrett**
Closes #14 

Adds optional logging to [TensorBoard](https://www.tensorflow.org/tensorboard), controlled by the `--tensorboard` cli arg.

Nice run names can be set with, for example, `--run-name experiment1`.

Start tensorboard with `tensorboard --logdir=runs`.

Please feel free to use as much or as little of this as you wish, or I'm happy to work on it if you'd like changes.

### CODE REVIEW

**Comment by rparrett** [comment]
The new dep probably needs to be added to the pypy project. I can add/test that when I get a moment, I haven't really been using it.

**Comment by hvoss-techfak** [comment]
Thank you so much for the changes. I'll merge the pull request and do a new release shortly.
